internal/db/repos/instance.go

```diff
@@ -76,13 +76,32 @@ func (r *InstanceRepository) UpdateStatusByName(ctx context.Context, name string
 		Update("status", status).Error
 }
 
-// List retrieves a paginated list of instances
+// List retrieves a paginated list of instances with optional status and soft delete filters
 func (r *InstanceRepository) List(ctx context.Context, opts *models.ListOptions) ([]models.Instance, error) {
 	var instances []models.Instance
 	query := r.db.WithContext(ctx)
 
-	if opts != nil && opts.IncludeDeleted {
-		query = query.Unscoped()
+	if opts != nil {
```

Comment:
similar to the client package, the opts are going to be used in multiple methods here. We should have a helper function that parses the opts into the db query so that we can unit test it.